extension/vector/src/index/hnsw_graph.cpp

```diff
@@ -68,10 +70,14 @@ float* OnDiskEmbeddings::getEmbedding(transaction::Transaction* transaction,
     KU_ASSERT(scanState.outputVectors.size() == 1 &&
               scanState.outputVectors[0]->state->getSelVector()[0] == 0);
     const auto value = scanState.outputVectors[0]->getValue<common::list_entry_t>(0);
-    KU_ASSERT(value.size == typeInfo.dimension);
+    KU_ASSERT(value.size == typeInfo.getNumElements());
     KU_UNUSED(value);
     const auto dataVector = common::ListVector::getDataVector(scanState.outputVectors[0]);
-    return reinterpret_cast<float*>(dataVector->getData()) + value.offset;
+    void* val = nullptr;
```

Comment:
Using void* is a bit unsafe, might be better using template there.

Reply:
This would require templating the whole class or most functions in the class. I was trying to avoid that so we can bloat functions less when we add more data types later. Though this might be worth revisiting later.